frontend/src/components/pages/teacher/ClassroomsPage.tsx

```diff
@@ -136,83 +67,95 @@ const ClassroomsPage = (): React.ReactElement => {
         <HeaderWithButton
           buttonText="Add New Classroom"
           onClick={handleAddClassroom}
-          showButton={classrooms.length !== 0}
+          showButton={classrooms?.length !== 0}
           title="Classroom"
         />
         <AddClassroomModal
           isOpen={isModalOpen}
           onClose={() => setIsModalOpen(false)}
         />
       </Box>
-      <Box flex="1">
-        {classrooms.length !== 0 ? (
-          <>
-            <Tabs index={tabIndex} marginTop={3} onChange={handleTabChange}>
-              <TabList>
-                <Tab color={unselectedTabColor}>Active</Tab>
-                <Tab color={unselectedTabColor}>Archived</Tab>
-              </TabList>
-              <TabPanels>
-                <TabPanel padding="0">
-                  <Grid gap={4} templateColumns="repeat(4, 1fr)">
-                    {paginatedData.map((classroom) => (
-                      <GridItem key={classroom.id} flex="1" paddingTop="4">
-                        <ClassroomCard
-                          key={classroom.id}
-                          activeAssessments={classroom.activeAssessments}
-                          assessmentCount={classroom.assessmentCount}
-                          grade={classroom.grade}
-                          name={classroom.name}
-                          studentCount={classroom.studentCount}
+      {loading && (
+        <Center flex="1" margin="15%">
+          <LoadingState />
+        </Center>
+      )}
+      {error && (
+        <Box height="100%" mt={10}>
+          <ErrorState />
+        </Box>
+      )}
+      {classrooms && !error && !loading && (
+        <Box flex="1">
+          {classrooms.length !== 0 ? (
+            <>
+              <Tabs index={tabIndex} marginTop={3} onChange={handleTabChange}>
+                <TabList>
+                  <Tab color={unselectedTabColor}>Active</Tab>
+                  <Tab color={unselectedTabColor}>Archived</Tab>
+                </TabList>
+                <TabPanels>
+                  <TabPanel padding="0">
+                    <Grid gap={4} templateColumns="repeat(4, 1fr)">
+                      {paginatedData?.map((classroom) => (
```

Comment:
nit: prefer destructuring:
```suggestion
                      {paginatedData?.map(({ gradeLevel, className }) => (
```